### PR TITLE
OCPQE-23148: Ensure that Azure HCs are created in the MC's location

### PR DESCRIPTION
**Pull request by fxierh**
We did the same to AWS. Now we do this to Azure as well. 

Thanks @jianlinliu for bringing this up. 

Successful rehearsals:
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/53560/rehearse-53560-periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-azure-aks-hypershift-guest-f7/1805134451110318080 
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/53560/rehearse-53560-periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-ipi-ovn-hypershift-etcd-encryption-guest-f14/1805134451093540864
https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/53560/rehearse-53560-periodic-ci-openshift-hypershift-release-4.16-periodics-e2e-azure-ovn-conformance/1805134450896408576

### CODE REVIEW

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.16-periodics-e2e-azure-ovn-conformance periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-ipi-ovn-hypershift-etcd-encryption-guest-f14 periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-azure-aks-hypershift-guest-f7

**Comment by fxierh** [comment]
/cc @sjenning @bryan-cox 

Could you please take a look ? Thank you.

File: ci-operator/step-registry/aks/provision/aks-provision-commands.sh [hunk 1]
```diff
@@ -21,6 +21,7 @@ AKE_CREATE_COMMAND=(
     --node-count "$AKS_NODE_COUNT"
     --load-balancer-sku "$AKS_LB_SKU"
     --os-sku "$AKS_OS_SKU"
+    --location "$LEASED_RESOURCE"
```
Comment (jianlinliu) [inline ci-operator/step-registry/aks/provision/aks-provision-commands.sh:24]: Accordingly shall we update aks-deprovision step too?

File: ci-operator/step-registry/hypershift/azure/create/hypershift-azure-create-chain.yaml [hunk 2]
```diff
@@ -15,7 +15,9 @@ chain:
       documentation: "Release image used for the HostedCluster. Empty by default it will use release:latest"
     - name: HYPERSHIFT_AZURE_LOCATION
       default: "eastus"
```
Comment (jianlinliu) [inline ci-operator/step-registry/hypershift/azure/create/hypershift-azure-create-chain.yaml:17]: We need to change the default value here, too, right?

**Comment by jianlinliu** [comment]
@fxierh shall we update `hypershift-setup-nested-management-cluster` together?

File: ci-operator/step-registry/aks/provision/aks-provision-commands.sh (hunk 1 above)
Comment (fxierh) [reply ci-operator/step-registry/aks/provision/aks-provision-commands.sh:24]: The `az aks delete` command does not have a flag to specify the location. It seems Azure automatically detects the location of the existing cluster for us.

File: ci-operator/step-registry/hypershift/azure/create/hypershift-azure-create-chain.yaml (hunk 2 above)
Comment (fxierh) [reply ci-operator/step-registry/hypershift/azure/create/hypershift-azure-create-chain.yaml:17]: I believe not. The dev team utilizes the hypershift cluster profile, which, unlike azure-qe, does not provide a meaningful value for `$LEASED_RESOURCE`. Keeping this value is necessary for the existing jobs set up by the dev team to function correctly.

File: ci-operator/step-registry/aks/provision/aks-provision-commands.sh (hunk 1 above)
Comment (jianlinliu) [reply ci-operator/step-registry/aks/provision/aks-provision-commands.sh:24]: got it.

**Comment by fxierh** [comment]
@jianlinliu 

> shall we update `hypershift-setup-nested-management-cluster` together?

I believe not. 

Unlike the `hypershift-azure-create` chain, `hypershift-setup-nested-management-cluster` is used exclusively by the dev team. Similar to https://github.com/openshift/release/pull/53560#discussion_r1652010332, it needs a meaningful default value and should not default to $LEASED_RESOURCE.

Thanks for the review.

**Pull request merged**